### PR TITLE
OG-418 prioritize paymaster pre rejection over acceptance budget

### PR DESCRIPTION
**Pull request by shahafn**

### CODE REVIEW

File: contracts/test/TestPaymasterConfigurableMisbehavior.sol [hunk 1]
```diff
@@ -44,6 +45,9 @@ contract TestPaymasterConfigurableMisbehavior is TestPaymasterEverythingAccepted
     function setExpensiveGasLimits(bool val) public {
         expensiveGasLimits = val;
     }
+    function setPreGasHigherThanAcceptance(bool val) public {
```
Comment (drortirosh) [inline contracts/test/TestPaymasterConfigurableMisbehavior.sol:48]: you have `setGasLimits` for this, setting all 3 gas params

File: contracts/test/TestPaymasterConfigurableMisbehavior.sol [hunk 2]
```diff
@@ -124,6 +128,13 @@ contract TestPaymasterConfigurableMisbehavior is TestPaymasterEverythingAccepted
             return IPaymaster.GasAndDataLimits(limits.acceptanceBudget * 9, limits.preRelayedCallGasLimit, limits.postRelayedCallGasLimit,
             limits.calldataSizeLimit);
         }
+        if (preHigherThanAcceptance) {
+            return IPaymaster.GasAndDataLimits(
+                limits.acceptanceBudget,
+                limits.preRelayedCallGasLimit + limits.acceptanceBudget + relayHub.calldataGasCost(limits.calldataSizeLimit),
```
Comment (drortirosh) [inline contracts/test/TestPaymasterConfigurableMisbehavior.sol:134]: too much magic in contract instead of in the test itself.

Comment (shahafn) [reply contracts/test/TestPaymasterConfigurableMisbehavior.sol:134]: Fixing, though this is a test contract, so does it matter where this magic happens?

File: contracts/test/TestPaymasterConfigurableMisbehavior.sol (hunk 1 above)
Comment (shahafn) [reply contracts/test/TestPaymasterConfigurableMisbehavior.sol]: fixed

**Review by drortirosh — APPROVED** [review]

**Pull request merged**